src/cryptoadvance/specter/server_endpoints/devices.py

```diff
@@ -124,22 +124,31 @@ def new_device():
                 device = app.specter.device_manager.add_device(
                     name=device_name, device_type=device_type, keys=[]
                 )
-                device.setup_device(file_password, app.specter.wallet_manager)
-                device.add_hot_wallet_keys(
-                    mnemonic,
-                    passphrase,
-                    paths,
-                    file_password,
-                    app.specter.wallet_manager,
-                    is_testnet(app.specter.chain),
-                    keys_range=[range_start, range_end],
-                    keys_purposes=keys_purposes,
-                )
-                flash("{} was added successfully!".format(device_name))
-                return redirect(
-                    url_for("devices_endpoint.device", device_alias=device.alias)
-                    + "?newdevice=true"
-                )
+                try:
+                    device.setup_device(file_password, app.specter.wallet_manager)
+                    device.add_hot_wallet_keys(
+                        mnemonic,
+                        passphrase,
+                        paths,
+                        file_password,
+                        app.specter.wallet_manager,
+                        is_testnet(app.specter.chain),
+                        keys_range=[range_start, range_end],
+                        keys_purposes=keys_purposes,
+                    )
+                    flash("{} was added successfully!".format(device_name))
+                    return redirect(
+                        url_for("devices_endpoint.device", device_alias=device.alias)
+                        + "?newdevice=true"
+                    )
+                except Exception as e:
+                    flash(f"Failed to setup hot wallet. Error: {e}", "error")
```

Comment:
I think the rule should be: If you catch a SpecterError, do that flash and move on. However if you catch an Exception, then you don't know what the real cause of that issue is, so you should make sure the stacktrace ends up in the log.
```suggestion
                    handle_exception(e)
                    flash(f"Failed to setup hot wallet. Error: {e}", "error")
```
(And import it)